Coverage

brettz9 opened the pull request:
Builds on #99.

Has one commit to add `nyc` to check coverage and another commit to bring the project to 100% coverage.

brettz9 commented:
I've rebased and also bumped the version to drop Node 6 (the latest nyc I added requires it, it also would allow using up-to-date versions of Mocha and `import-from`, and even Node 8 has been end-of-lifed and being dropped by many projects).

brettz9 commented:
The PR now also bumps peerDeps to support ESLint 7 so there is no warning about that.

brettz9 commented:
Thanks for the merge! I've rebased this one now as well...

brettz9 commented:
Was breakage with Mocha 8 and ESLint 3, so also dropped <=3 from peerDeps and Travis.

brettz9 commented:
And with Mocha 8 requiring Node 10 and Node 8 end-of-lifed, I added another commit to drop Node 8 from `engines`. If you still want the support, `travis.yml` should test Node 8 I think but it would also need to downgrade Mocha as well as ESLint.

Turbo87 commented:
@brettz9 can you split these changes into multiple PRs so that they can be discussed individually? 🙏

brettz9 commented:
Sure, I'm happy to split them up, and I started #114 which presents one item by itself.

---- 
FWIW though, if you are interested to know my process on why I connected all of these...

I added `nyc` so one can get reports that all lines of your code base are covered by the tests.  I set thresholds at 100% so any regressions of coverage would break Travis. And to avoid breaking Travis with such coverage now checked, I added tests necessary to bring the project to 100% coverage.

Since the latest `nyc` requires Node 8+, I updated `travis.yml` to drop Node 8 and check the current supported Node versions (12 and 14). And if you are ok upgrading, I figured we might as well also check ESLint 7 and Mocha 8 which depend on higher Node versions, though with Mocha 8, we have to drop ESLint 3 (as it fails with it) and bump Node further to Node 10+.
